### ISSUE TITLE
I ran sudo dlite install and now my mac is hosed

### ISSUE THREAD

**Issue by philipbergen** [nlf/dlite#41]
Something was changed with /var/empty/.../Application support

Now chrome, box and probably more apps are completely dead. Can't start.

I don't dare to reboot until I know what it did.

**Comment by philipbergen** [comment]
I have yosemite 10.10.5, it's an imac with hybrid drive.

**Comment by nlf** [comment]
whoa, that's pretty severe. how did you run it exactly? just `sudo dlite install`? from what shell? the installer doesn't _delete_ anything, and altering the files it does shouldn't have any impact like you're seeing. can you give me any more information at all?

**Comment by philipbergen** [comment]
Mail and all the other programs are dead too. It's like it wiped out application support. 
I can see the files are there (except /var/empty is indeed empty, what is that supposed to do?). 

I don't appear to have any accidentally root owned directories in my home.

It did this from zsh.

It started as soon as "Building disk image" appeared.

**Comment by philipbergen** [comment]
When I launch a new terminal all my zsh settings are gone.

**Comment by philipbergen** [comment]
I get a plain bash.

**Comment by philipbergen** [comment]
What exactly is it doing in that startup?

**Comment by philipbergen** [comment]
It has moved my home directory to /var/empty!!! Not the contents of home, but the variable setting that denotes the location of my home.

Where is that configured?

**Comment by philipbergen** [comment]
Where is that set?

**Comment by philipbergen** [comment]
Not the contents of home, but the variable setting that denotes the location of my home. In case that was unclear.

**Comment by nlf** [comment]
the only thing i can think of is that something in your shell corrupted the environment variables that `sudo` sets (i.e. $SUDO_USER, $SUDO_UID, $SUDO_GID) but even then it would only cause the installer to create a file in the wrong directory. nowhere in the code do i change any environment variables, so that shouldn't be possible.

can you run the `Console` app? if so can you try to poke around in there and see if there are any errors related to dlite?

**Comment by philipbergen** [comment]
Super quiet in the logs actually. This is all I have around that time

```
1/26/16 9:32:16.000 AM kernel[0]: VNOP_IOCTL(_DKIOCCSMAP) returned an unexpected error code=30
1/26/16 9:32:16.000 AM kernel[0]: VNOP_IOCTL(_DKIOCCSMAP) returned an unexpected error code=30
1/26/16 9:32:16.000 AM kernel[0]: VNOP_IOCTL(_DKIOCCSMAP) returned an unexpected error code=30
```

**Comment by philipbergen** [comment]
Just to clarify, I ran it just like that. Downloaded, chmod a+x, ran `sudo ~/Downloads/dlite install`

```
MD5 (/Users/pbergen/Downloads/dlite) = 8b84bcf7ee7b57c733dc61f618305c53
```

**Comment by philipbergen** [comment]
I have no SUDO_\* env vars at all.

**Comment by nlf** [comment]
the SUDO_ environment variables are only set when running a command through sudo, if you run `sudo env` you should see them.

doing some searching through OS X error codes, it sounds like error code 30 signals that a disk volume was taken offline and the OS wants it to be reinserted. you might try running `sudo dmesg` and reading through output, but i suspect that your physical disk may be going bad and was taken offline as a result of that. can you run disk utility? if so, do all of your volumes show as currently online and mounted?

**Comment by philipbergen** [comment]
So the problem is with Active Directory. It somehow killed off my network account. I'm working on figuring out the repair for it.

You might want to check what could possibly have caused that in your installer.

**Comment by nlf** [comment]
i don't do anything with the user's account.. i create a few files in their home directory and set the ownership of those to the user, but that's it. i have no idea what on earth could've caused your account to be removed. is your home directory on some sort of shared volume? maybe creating a 30GiB file over the network to a shared volume caused the issue

**Comment by nlf** [comment]
i'm going to close this issue, but please do let me know if you find out anything more about why this may have happened.

**Comment by zenoamaro** [comment]
Whenever I launched `dlite install` it would simply hang the whole system. No disk access was possible anymore, and any process that tried to would become a zombie immediately.

I noticed that I also have a hybrid drive. Keeping _console.app_ open during the install showed me that _core storage_ was failing repeatedly, reporting something along the lines of _"cannot move data, err code = 30"_. Of course all those log messages have not been flushed to disk, along with a few modifications I did just before launching, so I cannot post them here at the moment.

Interestingly enough, when trying to build a 5gb image nothing of the sort happened. Building a 10gb image causes the issue immediately. I'll try to make a video of that happening as soon as I can.

**Comment by zenoamaro** [comment]
Correction, it does happen even when creating a 5gb image with `-d 5`, consistently. Every time I have to force shutdown the mac and reboot. I guess I had a single lucky run earlier, but I cannot see what the difference could be.

Anyway, here is the a screenshot of console.app. I put the marker just before starting `sudo dlite install`. As soon as the first message appears, the mac is unusable for anything that requires disk. After a while even _console.app_ hangs. The only solution is to force reboot via long press of the power button.

![20160202_125755](https://cloud.githubusercontent.com/assets/493139/12749011/3d777940-c9ae-11e5-8669-19d448351c97.jpg)

**Comment by rjayako** [comment]
@zenoamaro any updates? my iMac keeps crashing too :(

**Comment by zenoamaro** [comment]
@rjayako no updates, I checked again a few weeks ago but it would still hose ~90% of the time.

**Comment by quadule** [comment]
I'm not a dlite user but I've had my system (also iMac / Yosemite / Fusion Drive) hosed by this same issue several times last year after creating large files (downloads in Transmission in my case). Tried a full reformat and reinstall twice and watched it happen again each time with the same error messages in the logs, which I've been Googling every once in a while ever since.

I had the exact same symptoms and log messages as reported by @philipbergen and @zenoamaro, but I'd also get a flood of "Keychain Not Found" error dialogs. After that my login.keychain and seemingly every other file that was in use at the time was truncated to 0 bytes.

Neither Apple Hardware Test nor the Genius Bar could find any issues so I eventually tried splitting the Fusion Drive before reinstalling and I haven't had any issues since. I also upgraded to El Capitan shortly after this, [but I don't think that had anything to do with it](https://discussions.apple.com/thread/7486970). As far as I can tell, that thread and this one are the only search results for these error messages.
